### DISCUSSION TITLE
Add touch events

### DISCUSSION THREAD

**Discussion by SleepyF0X** [hardkoded/puppeteer-sharp#2379] [Ideas]
Hi, i wanna to add touch events to your repo. I want to implement logic from https://github.com/puppeteer/puppeteer/blob/main/packages/puppeteer-core/src/cdp/Input.ts
Also i have code for swipe (touchStart + touchMove + touchEnd), this use 4 coordinates, startX, startY, moveX and moveY.
Now i use reflection in my project, and use this events manually, but i want to use it from your package, and it will be great if you allow this to be realized.
I haven’t yet figured out how to add testing to your project for these events, but I’ll figure it out if you allow

**Comment by SleepyF0X** [comment]
This is helpfull, if user want to scroll by swipe. 
Also, now I realised functionality for scroll by mouse wheel, and swipes. These functions look more human than simply executing a script
IDK, maybe i can also add it. Because if we choose desktop device - we want to use mouse wheel, if mobile - swipes.

**Comment by kblok** [comment]
Hey! You are welcome to introduce this feature! You will need to implement the [should report touchMove](https://github.com/puppeteer/puppeteer/commit/c8bb11adfcf1537032730a91baa3c36a6e324926#diff-b5da00b39c4349af55cfd89e2dc8752d22d033d9ba831447d27963485d5ceaefR56) test. Let me know if you need some help :)
